subprojects/docs/src/docs/release/notes.md

```diff
@@ -6,6 +6,28 @@ Here are the new features introduced in this Gradle release.
 IMPORTANT: if this is a patch release, ensure that a prominent link is included in the foreword to all releases of the same minor stream.
 Add-->
 
+### Timeouts for HTTP requests
+
+Previous versions of Gradle did not define a timeout for any HTTP requests. Under certain conditions e.g. network problems, unresponsive or overloaded servers this behavior could lead to hanging connections.
+Gradle now defines connection and socket timeouts for all HTTP requests. In the event of a timeout (or any `IOException`s which are not caused by error HTTP status code), Gradle will skip subsequent connections to the same repository for the duration of the build. 
+The output of a build clearly indicates which request was skipped.
+
+```
+* What went wrong:
+Could not resolve all files for configuration ':deps'.
+> Could not resolve group:a:1.0.
+  Required by:
+      project :
+   > Could not resolve group:a:1.0.
+      > Could not get resource 'http://localhost:54347/repo/group/a/1.0/a-1.0.pom'.
+         > Could not GET 'http://localhost:54347/repo/group/a/1.0/a-1.0.pom'.
+            > Read timed out
+> Could not resolve group:b:1.0.
+  Required by:
+      project :
+   > Skipped due to earlier error
+```
+
```

Comment:
@blindpirate Could you also include that the changes are effective for the build cache, too. Something like this: https://github.com/gradle/gradle/blame/3f57fc089f2b983c14d7b3f5d2a2e62f42ab6651/subprojects/docs/src/docs/release/notes.md#L131-L140

Reply:
Ah, my fault, will do it now.

Reply:
@wolfs Done. Thanks for reminding.

Reply:
Thank you!

Reply:
My pleasure!